src/TensorFlowNET.Keras/Engine/Model.Evaluate.cs

```diff
@@ -46,18 +55,32 @@ public partial class Model
                 StepsPerExecution = _steps_per_execution
             });
 
+            var callbacks = new CallbackList(new CallbackParams
+            {
+                Model = this,
+                Verbose = verbose,
+                Steps = data_handler.Inferredsteps
+            });
+            callbacks.on_test_begin();
+
             foreach (var (epoch, iterator) in data_handler.enumerate_epochs())
             {
                 reset_metrics();
-                // callbacks.on_epoch_begin(epoch)
+                //callbacks.on_epoch_begin(epoch);
                 // data_handler.catch_stop_iteration();
-                IEnumerable<(string, Tensor)> results = null;
+                IEnumerable<(string, Tensor)> logs = null;
+
                 foreach (var step in data_handler.steps())
                 {
-                    // callbacks.on_train_batch_begin(step)
-                    results = test_function(data_handler, iterator);
+                    callbacks.on_test_batch_begin(step);
+                    logs = test_function(data_handler, iterator);
+                    var end_step = step + data_handler.StepIncrement;
+                    callbacks.on_test_batch_end(end_step, logs);
                 }
             }
+            Console.WriteLine();
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
         }
 
         public KeyValuePair<string, float>[] evaluate(IDatasetV2 x)
```

Comment:
This method is an overload of the method above, which should have a similar implementation. I think maybe adding an `EvaluateInternal` is better, just like `FitInternal`.